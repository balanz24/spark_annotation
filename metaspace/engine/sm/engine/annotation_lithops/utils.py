from __future__ import annotations

import json
import logging
from base64 import urlsafe_b64encode
from hashlib import blake2b

logger = logging.getLogger('annotation-pipeline')


def jsonhash(obj) -> str:
    """
    Calculates a hash for a JSON-stringifiable object. Intended for compacting large sets of
    parameters into a simple key that can be used to distinctly identify a cache entry.

    The output is collision-resistant, but shouldn't be assumed to be cryptographically secure.
    In most cases a motivated adversary could figure out the original object contents easily, as
    there's no hidden key and it's unlikely there will be much variety in the objects hashed.
    """
    json_val = json.dumps(obj, sort_keys=True)
    hash_val = blake2b(json_val.encode(), digest_size=12).digest()
    return str(urlsafe_b64encode(hash_val), 'utf-8')
