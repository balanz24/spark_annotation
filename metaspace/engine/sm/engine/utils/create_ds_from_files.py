import json
from datetime import datetime
from pathlib import Path

from sm.engine import molecular_db
from sm.engine.dataset import Dataset


def create_ds_from_files(ds_id, ds_name, ds_input_path, config_path=None, meta_path=None):
    config_path = config_path or Path(ds_input_path) / 'config.json'
    ds_config = json.load(open(config_path))
    if 'database_ids' not in ds_config:
        ds_config['database_ids'] = [
            molecular_db.find_by_name(db, True).id for db in ds_config['databases']
        ]

    meta_path = meta_path or Path(ds_input_path) / 'meta.json'
    if not Path(meta_path).exists():
        raise Exception('meta.json not found')
    metadata = json.load(open(str(meta_path)))

    return Dataset(
        id=ds_id,
        name=ds_name,
        input_path=str(ds_input_path),
        upload_dt=datetime.now(),
        metadata=metadata,
        is_public=True,
        config=ds_config,
    )
