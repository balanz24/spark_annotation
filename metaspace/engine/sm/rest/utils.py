import json

import bottle

OK = {'status_code': 200, 'status': 'success'}
NOT_EXIST = {'status_code': 404, 'status': 'not_exist'}
ALREADY_EXISTS = {'status_code': 400, 'status': 'already_exists'}
WRONG_PARAMETERS = {'status_code': 400, 'status': 'wrong_parameters'}
INTERNAL_ERROR = {'status_code': 500, 'status': 'server_error'}


def body_to_json(request):
    body = request.body.getvalue()
    return json.loads(body.decode('utf-8'))


def make_response(status_doc, **kwargs):
    bottle.response.status = status_doc['status_code']
    return {'status': status_doc['status'], **kwargs}
