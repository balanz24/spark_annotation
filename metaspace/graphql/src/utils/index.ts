export { createConnection, findUserByEmail } from './db'
export { smApiDatasetRequest } from './smApi/datasets'

export type LooselyCompatible<T> =
  {[K in keyof T]?: T[K] extends (string | number | null | undefined) ? T[K] | null : any};
