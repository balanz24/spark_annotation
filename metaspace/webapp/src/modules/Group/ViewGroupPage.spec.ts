import { mount, Stubs } from '@vue/test-utils'
import Vue from 'vue'
import ViewGroupPage from './ViewGroupPage.vue'
import router from '../../router'
import { initMockGraphqlClient, apolloProvider } from '../../../tests/utils/mockGraphqlClient'

describe('ViewGroupPage', () => {
  const mockMembersForAdmins = [
    {
      role: 'GROUP_ADMIN',
      numDatasets: 123,
      user: { id: '3', name: 'me', email: 'my-email@example.com' },
    },
    {
      role: 'PENDING',
      numDatasets: 0,
      user: { id: '4', name: 'Person who asked to join', email: 'access@requestor.com' },
    },
    {
      role: 'INVITED',
      numDatasets: 0,
      user: { id: '5', name: 'Invitee', email: 'awaiting@response.com' },
    },
    {
      role: 'MEMBER',
      numDatasets: 1,
      user: { id: '6', name: 'Project member', email: 'person@embl.de' },
    },
  ]
  const mockMembersForMembers = mockMembersForAdmins.map(m => ({ ...m, user: { ...m.user, email: null } }))
  const mockMembersForPublic = mockMembersForMembers.filter(m => m.role === 'GROUP_ADMIN')

  const mockGroup = {
    id: '00000000-1111-2222-3333-444444444444',
    name: 'group name',
    shortName: 'groupShortName',
    urlSlug: null,
    currentUserRole: null,
    numMembers: 2,
    members: mockMembersForPublic,
    numDatabases: 2,
  }

  const mockGroupFn = jest.fn((src: any, args: any, ctx: any, info: any): any => mockGroup)
  const graphqlMocks = {
    Query: () => ({
      currentUser: () => ({ id: 'userid' }),
      group: mockGroupFn,
      groupByUrlSlug: mockGroupFn,
      allDatasets: () => ([
        { id: 'datasetId1', name: 'dataset name 1', status: 'FINISHED' },
        { id: 'datasetId2', name: 'dataset name 2', status: 'QUEUED' },
        { id: 'datasetId3', name: 'dataset name 3', status: 'ANNOTATING' },
        { id: 'datasetId4', name: 'dataset name 4', status: 'FINISHED' },
      ]),
      countDatasets: () => 4,
    }),
  }

  const stubs: Stubs = {
    DatasetItem: true,
    MolecularDatabases: true,
  }
  const stubsWithMembersList: Stubs = {
    ...stubs,
    GroupMembersList: true,
  }

  beforeEach(() => {
    jest.clearAllMocks()
    router.replace({ name: 'group', params: { groupIdOrSlug: mockGroup.id } })
  })

  describe('datasets tab', () => {
    beforeEach(() => {
      router.replace({ query: { tab: 'datasets' } })
    })

    it('should match snapshot (non-member)', async() => {
      initMockGraphqlClient(graphqlMocks)
      const maxVisibleDatasets = 3
      const wrapper = mount(ViewGroupPage, { router, stubs, apolloProvider })
      wrapper.setData({ maxVisibleDatasets }) // Also test that the datasets list is correctly clipped
      await Vue.nextTick()

      expect(wrapper).toMatchSnapshot()

      expect(wrapper.findAll('button').wrappers.map(w => w.text()))
        .toEqual(expect.arrayContaining(['Request access']))
      expect(wrapper.findAll('[role="tab"]').wrappers.map(w => w.text()))
        .toEqual(['Description', 'Datasets (4)', 'Members (2)'])
      expect(wrapper.findAll('.dataset-list > *')).toHaveLength(maxVisibleDatasets)
    })
  })

  describe('members tab', () => {
    beforeEach(() => {
      router.replace({ query: { tab: 'members' } })
    })

    it('should match snapshot (non-member)', async() => {
      initMockGraphqlClient(graphqlMocks)
      const wrapper = mount(ViewGroupPage, { router, stubs: stubsWithMembersList, apolloProvider })
      await Vue.nextTick()

      expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot (invited)', async() => {
      mockGroupFn.mockImplementation(() => ({ ...mockGroup, currentUserRole: 'INVITED' }))
      initMockGraphqlClient(graphqlMocks)
      const wrapper = mount(ViewGroupPage, { router, stubs: stubsWithMembersList, apolloProvider })
      await Vue.nextTick()

      expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot (member)', async() => {
      mockGroupFn.mockImplementation(() =>
        ({ ...mockGroup, currentUserRole: 'MEMBER', members: mockMembersForMembers }))
      initMockGraphqlClient(graphqlMocks)
      const wrapper = mount(ViewGroupPage, { router, stubs: stubsWithMembersList, apolloProvider })
      await Vue.nextTick()

      expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot (manager, including table)', async() => {
      mockGroupFn.mockImplementation(() =>
        ({ ...mockGroup, currentUserRole: 'GROUP_ADMIN', members: mockMembersForAdmins }))
      initMockGraphqlClient(graphqlMocks)
      const wrapper = mount(ViewGroupPage, { router, stubs, apolloProvider })
      await Vue.nextTick()

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('settings tab', () => {
    beforeEach(() => {
      router.replace({ query: { tab: 'settings' } })
    })

    it('should match snapshot', async() => {
      mockGroupFn.mockImplementation(() => ({ ...mockGroup, currentUserRole: 'GROUP_ADMIN' }))
      initMockGraphqlClient(graphqlMocks)
      const wrapper = mount(ViewGroupPage, { router, stubs, apolloProvider })
      await Vue.nextTick()

      expect(wrapper).toMatchSnapshot()
    })
  })

  it('should correctly fetch data and update the route if the group has a urlSlug but is accessed by ID', async() => {
    const urlSlug = 'group-url-slug'
    mockGroupFn.mockImplementation(() => ({ ...mockGroup, urlSlug }))
    initMockGraphqlClient(graphqlMocks)

    const wrapper = mount(ViewGroupPage, { router, stubs, apolloProvider })
    await Vue.nextTick()

    expect(router.currentRoute.params.groupIdOrSlug).toEqual(urlSlug)
    // Assert that the correct graphql calls were made. See the GraphQLFieldResolver type for details on the args here
    expect(mockGroupFn).toHaveBeenCalledTimes(2)
    expect(mockGroupFn.mock.calls[0][1].groupId).toEqual(mockGroup.id)
    expect(mockGroupFn.mock.calls[0][3].fieldName).toEqual('group')
    expect(mockGroupFn.mock.calls[1][1].urlSlug).toEqual(urlSlug)
    expect(mockGroupFn.mock.calls[1][3].fieldName).toEqual('groupByUrlSlug')
  })
})
