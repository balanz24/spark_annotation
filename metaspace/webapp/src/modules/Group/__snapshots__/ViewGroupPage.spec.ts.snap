// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewGroupPage datasets tab should match snapshot (non-member) 1`] = `
<div class="page" mock-v-loading="false">
  <div class="page-content">
    <!---->
    <transition-stub name="dialog-fade" class="requested-access-dialog">
      <div class="el-dialog__wrapper" style="display: none;">
        <div role="dialog" aria-modal="true" aria-label="group name Access Request" class="el-dialog" style="margin-top: 15vh;">
          <div class="el-dialog__header"><span class="el-dialog__title">group name Access Request</span><button type="button" aria-label="Close" class="el-dialog__headerbtn"><i class="el-dialog__close el-icon el-icon-close"></i></button></div>
          <!---->
          <!---->
        </div>
      </div>
    </transition-stub>
    <div class="header-row">
      <div class="header-names">
        <h1>group name</h1>
        <h2 class="ml-3 text-gray-600">
          (groupShortName)
        </h2>
      </div>
      <div class="header-buttons"><button type="button" class="el-button el-button--primary">
          <!---->
          <!----><span>
          Request access
        </span></button>
        <!---->
      </div>
      <!---->
    </div>
    <div class="with-badges el-tabs el-tabs--top">
      <div class="el-tabs__header is-top">
        <div class="el-tabs__nav-wrap is-top">
          <div class="el-tabs__nav-scroll">
            <div role="tablist" class="el-tabs__nav is-top" style="transform: translateX(-0px);">
              <div class="el-tabs__active-bar is-top" style="width: 0px; transform: translateX(NaNpx); webkit-transform: translateX(NaNpx);"></div>
              <div id="tab-description" aria-controls="pane-description" role="tab" tabindex="-1" class="el-tabs__item is-top">Description</div>
              <div id="tab-datasets" aria-controls="pane-datasets" role="tab" aria-selected="true" tabindex="0" class="el-tabs__item is-top is-active">Datasets (4)</div>
              <div id="tab-members" aria-controls="pane-members" role="tab" tabindex="-1" class="el-tabs__item is-top"><span>
          Members (2)
          <!----></span></div>
            </div>
          </div>
        </div>
      </div>
      <div class="el-tabs__content">
        <!---->
        <div role="tabpanel" id="pane-datasets" aria-labelledby="tab-datasets" class="el-tab-pane">
          <div class="dataset-list">
            <datasetitem-stub dataset="[object Object]" current-user="[object Object]" idx="0" hide-group-menu="true"></datasetitem-stub>
            <datasetitem-stub dataset="[object Object]" current-user="[object Object]" idx="1" hide-group-menu="true"></datasetitem-stub>
            <datasetitem-stub dataset="[object Object]" current-user="[object Object]" idx="2" hide-group-menu="true"></datasetitem-stub>
            <!---->
          </div>
          <div class="dataset-list-footer"><a href="/datasets?grp=00000000-1111-2222-3333-444444444444" class="">
              See all datasets
            </a></div>
        </div>
        <!---->
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`ViewGroupPage members tab should match snapshot (invited) 1`] = `
<div class="page" mock-v-loading="false">
  <div class="page-content">
    <!---->
    <transition-stub name="dialog-fade" class="requested-access-dialog">
      <div class="el-dialog__wrapper" style="display: none;">
        <div role="dialog" aria-modal="true" aria-label="group name Access Request" class="el-dialog" style="margin-top: 15vh;">
          <div class="el-dialog__header"><span class="el-dialog__title">group name Access Request</span><button type="button" aria-label="Close" class="el-dialog__headerbtn"><i class="el-dialog__close el-icon el-icon-close"></i></button></div>
          <!---->
          <!---->
        </div>
      </div>
    </transition-stub>
    <div class="header-row">
      <div class="header-names">
        <h1>group name</h1>
        <h2 class="ml-3 text-gray-600">
          (groupShortName)
        </h2>
      </div>
      <div class="header-buttons">
        <!---->
        <!---->
      </div>
      <transition-stub name="el-alert-fade">
        <div role="alert" class="el-alert el-alert--info is-light"><i class="el-alert__icon el-icon-info is-big"></i>
          <div class="el-alert__content">
            <!---->
            <p class="el-alert__description">
              <div style="padding: 0px 0px 20px 20px;">
                <p>
                  You have been invited to join group name.
                </p>
                <div><button type="button" class="el-button el-button--danger">
                    <!---->
                    <!----><span>
              Decline invitation
            </span></button> <button type="button" class="el-button el-button--primary">
                    <!---->
                    <!----><span>
              Join group
            </span></button></div>
              </div>
            </p>
            <!----><i class="el-alert__closebtn el-icon-close" style="display: none;"></i>
          </div>
        </div>
      </transition-stub>
    </div>
    <div class="with-badges el-tabs el-tabs--top">
      <div class="el-tabs__header is-top">
        <div class="el-tabs__nav-wrap is-top">
          <div class="el-tabs__nav-scroll">
            <div role="tablist" class="el-tabs__nav is-top" style="transform: translateX(-0px);">
              <div class="el-tabs__active-bar is-top" style="width: 0px; transform: translateX(NaNpx); webkit-transform: translateX(NaNpx);"></div>
              <div id="tab-description" aria-controls="pane-description" role="tab" tabindex="-1" class="el-tabs__item is-top">Description</div>
              <div id="tab-datasets" aria-controls="pane-datasets" role="tab" tabindex="-1" class="el-tabs__item is-top">Datasets (4)</div>
              <div id="tab-members" aria-controls="pane-members" role="tab" aria-selected="true" tabindex="0" class="el-tabs__item is-top is-active"><span>
          Members (2)
          <!----></span></div>
            </div>
          </div>
        </div>
      </div>
      <div class="el-tabs__content">
        <!---->
        <!---->
        <div role="tabpanel" id="pane-members" aria-labelledby="tab-members" class="el-tab-pane">
          <groupmemberslist-stub currentuser="[object Object]" group="[object Object]" members="[object Object]" refreshdata="function () { [native code] }"></groupmemberslist-stub>
          <p class="hidden-members-text">
            + 1 hidden member.
          </p>
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`ViewGroupPage members tab should match snapshot (manager, including table) 1`] = `
<div class="page" mock-v-loading="false">
  <div class="page-content">
    <!---->
    <transition-stub name="dialog-fade" class="requested-access-dialog">
      <div class="el-dialog__wrapper" style="display: none;">
        <div role="dialog" aria-modal="true" aria-label="group name Access Request" class="el-dialog" style="margin-top: 15vh;">
          <div class="el-dialog__header"><span class="el-dialog__title">group name Access Request</span><button type="button" aria-label="Close" class="el-dialog__headerbtn"><i class="el-dialog__close el-icon el-icon-close"></i></button></div>
          <!---->
          <!---->
        </div>
      </div>
    </transition-stub>
    <div class="header-row">
      <div class="header-names">
        <h1>group name</h1>
        <h2 class="ml-3 text-gray-600">
          (groupShortName)
        </h2>
      </div>
      <div class="header-buttons">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
    <div class="with-badges el-tabs el-tabs--top">
      <div class="el-tabs__header is-top">
        <div class="el-tabs__nav-wrap is-top">
          <div class="el-tabs__nav-scroll">
            <div role="tablist" class="el-tabs__nav is-top" style="transform: translateX(-0px);">
              <div class="el-tabs__active-bar is-top" style="width: 0px; transform: translateX(NaNpx); webkit-transform: translateX(NaNpx);"></div>
              <div id="tab-description" aria-controls="pane-description" role="tab" tabindex="-1" class="el-tabs__item is-top">Description</div>
              <div id="tab-datasets" aria-controls="pane-datasets" role="tab" tabindex="-1" class="el-tabs__item is-top">Datasets (4)</div>
              <div id="tab-members" aria-controls="pane-members" role="tab" aria-selected="true" tabindex="0" class="el-tabs__item is-top is-active"><span>
          Members (2)
          <span class="notification"></span></span></div>
              <div id="tab-databases" aria-controls="pane-databases" role="tab" tabindex="-1" class="el-tabs__item is-top"><span>
            Databases (2)
          </span></div>
              <div id="tab-settings" aria-controls="pane-settings" role="tab" tabindex="-1" class="el-tabs__item is-top">Settings</div>
            </div>
          </div>
        </div>
      </div>
      <div class="el-tabs__content">
        <!---->
        <!---->
        <div role="tabpanel" id="pane-members" aria-labelledby="tab-members" class="el-tab-pane">
          <div>
            <!---->
            <div class="el-table el-table--fit el-table--scrollable-x el-table--enable-row-hover el-table--enable-row-transition" element-loading-text="Loading results from the server..." mock-v-loading="false">
              <div class="hidden-columns">
                <div></div>
                <div></div>
                <div></div>
                <div></div>
                <div></div>
              </div>
              <div class="el-table__header-wrapper">
                <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 850px;">
                  <colgroup>
                    <col name="el-table_1_column_1" width="200">
                    <col name="el-table_1_column_2" width="200">
                    <col name="el-table_1_column_3" width="160">
                    <col name="el-table_1_column_4" width="90">
                    <col name="el-table_1_column_5" width="200">
                  </colgroup>
                  <thead class="">
                    <tr class="">
                      <th colspan="1" rowspan="1" class="el-table_1_column_1     is-leaf">
                        <div class="cell">Name</div>
                      </th>
                      <th colspan="1" rowspan="1" class="el-table_1_column_2     is-leaf">
                        <div class="cell">Email</div>
                      </th>
                      <th colspan="1" rowspan="1" class="el-table_1_column_3     is-leaf">
                        <div class="cell">Role</div>
                      </th>
                      <th colspan="1" rowspan="1" class="el-table_1_column_4  is-center   is-leaf">
                        <div class="cell">Datasets</div>
                      </th>
                      <th colspan="1" rowspan="1" class="el-table_1_column_5     is-leaf">
                        <div class="cell"></div>
                      </th>
                    </tr>
                  </thead>
                </table>
              </div>
              <div class="el-table__body-wrapper is-scrolling-left">
                <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 850px;">
                  <colgroup>
                    <col name="el-table_1_column_1" width="200">
                    <col name="el-table_1_column_2" width="200">
                    <col name="el-table_1_column_3" width="160">
                    <col name="el-table_1_column_4" width="90">
                    <col name="el-table_1_column_5" width="200">
                  </colgroup>
                  <tbody>
                    <tr class="el-table__row">
                      <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                        <div class="cell">
                          me
                        </div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                        <div class="cell">
                          my-email@example.com
                        </div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                        <div class="cell"><a href="#" title="Change role">
                            Group admin
                          </a></div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_4 is-center ">
                        <div class="cell"><a href="/datasets?grp=00000000-1111-2222-3333-444444444444&amp;subm=3" class="">
                            123
                          </a></div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                        <div class="cell">
                          <!---->
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </td>
                    </tr>
                    <tr class="el-table__row">
                      <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                        <div class="cell">
                          Person who asked to join
                        </div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                        <div class="cell">
                          access@requestor.com
                        </div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                        <div class="cell"><span>
          Requesting access
        </span></div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_4 is-center ">
                        <div class="cell"><a href="/datasets?grp=00000000-1111-2222-3333-444444444444&amp;subm=4" class="">
                            0
                          </a></div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                        <div class="cell">
                          <!---->
                          <!----> <button type="button" class="el-button grid-button el-button--success el-button--mini">
                            <!---->
                            <!----><span>
          Accept
        </span></button> <button type="button" class="el-button grid-button el-button--default el-button--mini">
                            <!---->
                            <!----><span>
          Decline
        </span></button></div>
                      </td>
                    </tr>
                    <tr class="el-table__row">
                      <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                        <div class="cell">
                          Invitee
                        </div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                        <div class="cell">
                          awaiting@response.com
                        </div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                        <div class="cell"><span>
          Invited
        </span></div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_4 is-center ">
                        <div class="cell"><a href="/datasets?grp=00000000-1111-2222-3333-444444444444&amp;subm=5" class="">
                            0
                          </a></div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                        <div class="cell">
                          <!----> <button type="button" class="el-button grid-button el-button--default el-button--mini">
                            <!---->
                            <!----><span>
          Cancel
        </span></button>
                          <!---->
                          <!---->
                        </div>
                      </td>
                    </tr>
                    <tr class="el-table__row">
                      <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                        <div class="cell">
                          Project member
                        </div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                        <div class="cell">
                          person@embl.de
                        </div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                        <div class="cell"><a href="#" title="Change role">
                            Member
                          </a></div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_4 is-center ">
                        <div class="cell"><a href="/datasets?grp=00000000-1111-2222-3333-444444444444&amp;subm=6" class="">
                            1
                          </a></div>
                      </td>
                      <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                        <div class="cell"><button type="button" class="el-button grid-button el-button--default el-button--mini">
                            <!---->
                            <!----><span>
          Remove
        </span></button>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </td>
                    </tr>
                    <!---->
                  </tbody>
                </table>
                <!---->
                <!---->
              </div>
              <!---->
              <!---->
              <!---->
              <!---->
              <div class="el-table__column-resize-proxy" style="display: none;"></div>
            </div>
            <div class="pagination-row">
              <!---->
              <div class="flex-spacer"></div> <button type="button" class="el-button el-button--default">
                <!---->
                <!----><span>
      Add member
    </span></button>
            </div>
          </div>
          <!---->
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`ViewGroupPage members tab should match snapshot (member) 1`] = `
<div class="page" mock-v-loading="false">
  <div class="page-content">
    <!---->
    <transition-stub name="dialog-fade" class="requested-access-dialog">
      <div class="el-dialog__wrapper" style="display: none;">
        <div role="dialog" aria-modal="true" aria-label="group name Access Request" class="el-dialog" style="margin-top: 15vh;">
          <div class="el-dialog__header"><span class="el-dialog__title">group name Access Request</span><button type="button" aria-label="Close" class="el-dialog__headerbtn"><i class="el-dialog__close el-icon el-icon-close"></i></button></div>
          <!---->
          <!---->
        </div>
      </div>
    </transition-stub>
    <div class="header-row">
      <div class="header-names">
        <h1>group name</h1>
        <h2 class="ml-3 text-gray-600">
          (groupShortName)
        </h2>
      </div>
      <div class="header-buttons">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
    <div class="with-badges el-tabs el-tabs--top">
      <div class="el-tabs__header is-top">
        <div class="el-tabs__nav-wrap is-top">
          <div class="el-tabs__nav-scroll">
            <div role="tablist" class="el-tabs__nav is-top" style="transform: translateX(-0px);">
              <div class="el-tabs__active-bar is-top" style="width: 0px; transform: translateX(NaNpx); webkit-transform: translateX(NaNpx);"></div>
              <div id="tab-description" aria-controls="pane-description" role="tab" tabindex="-1" class="el-tabs__item is-top">Description</div>
              <div id="tab-datasets" aria-controls="pane-datasets" role="tab" tabindex="-1" class="el-tabs__item is-top">Datasets (4)</div>
              <div id="tab-members" aria-controls="pane-members" role="tab" aria-selected="true" tabindex="0" class="el-tabs__item is-top is-active"><span>
          Members (2)
          <span class="notification"></span></span></div>
              <div id="tab-databases" aria-controls="pane-databases" role="tab" tabindex="-1" class="el-tabs__item is-top"><span>
            Databases (2)
          </span></div>
            </div>
          </div>
        </div>
      </div>
      <div class="el-tabs__content">
        <!---->
        <!---->
        <div role="tabpanel" id="pane-members" aria-labelledby="tab-members" class="el-tab-pane">
          <groupmemberslist-stub currentuser="[object Object]" group="[object Object]" members="[object Object],[object Object],[object Object],[object Object]" refreshdata="function () { [native code] }"></groupmemberslist-stub>
          <!---->
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`ViewGroupPage members tab should match snapshot (non-member) 1`] = `
<div class="page" mock-v-loading="false">
  <div class="page-content">
    <!---->
    <transition-stub name="dialog-fade" class="requested-access-dialog">
      <div class="el-dialog__wrapper" style="display: none;">
        <div role="dialog" aria-modal="true" aria-label="group name Access Request" class="el-dialog" style="margin-top: 15vh;">
          <div class="el-dialog__header"><span class="el-dialog__title">group name Access Request</span><button type="button" aria-label="Close" class="el-dialog__headerbtn"><i class="el-dialog__close el-icon el-icon-close"></i></button></div>
          <!---->
          <!---->
        </div>
      </div>
    </transition-stub>
    <div class="header-row">
      <div class="header-names">
        <h1>group name</h1>
        <h2 class="ml-3 text-gray-600">
          (groupShortName)
        </h2>
      </div>
      <div class="header-buttons"><button type="button" class="el-button el-button--primary">
          <!---->
          <!----><span>
          Request access
        </span></button>
        <!---->
      </div>
      <!---->
    </div>
    <div class="with-badges el-tabs el-tabs--top">
      <div class="el-tabs__header is-top">
        <div class="el-tabs__nav-wrap is-top">
          <div class="el-tabs__nav-scroll">
            <div role="tablist" class="el-tabs__nav is-top" style="transform: translateX(-0px);">
              <div class="el-tabs__active-bar is-top" style="width: 0px; transform: translateX(NaNpx); webkit-transform: translateX(NaNpx);"></div>
              <div id="tab-description" aria-controls="pane-description" role="tab" tabindex="-1" class="el-tabs__item is-top">Description</div>
              <div id="tab-datasets" aria-controls="pane-datasets" role="tab" tabindex="-1" class="el-tabs__item is-top">Datasets (4)</div>
              <div id="tab-members" aria-controls="pane-members" role="tab" aria-selected="true" tabindex="0" class="el-tabs__item is-top is-active"><span>
          Members (2)
          <!----></span></div>
            </div>
          </div>
        </div>
      </div>
      <div class="el-tabs__content">
        <!---->
        <!---->
        <div role="tabpanel" id="pane-members" aria-labelledby="tab-members" class="el-tab-pane">
          <groupmemberslist-stub currentuser="[object Object]" group="[object Object]" members="[object Object]" refreshdata="function () { [native code] }"></groupmemberslist-stub>
          <p class="hidden-members-text">
            + 1 hidden member.
          </p>
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`ViewGroupPage settings tab should match snapshot 1`] = `
<div class="page" mock-v-loading="false">
  <div class="page-content">
    <!---->
    <transition-stub name="dialog-fade" class="requested-access-dialog">
      <div class="el-dialog__wrapper" style="display: none;">
        <div role="dialog" aria-modal="true" aria-label="group name Access Request" class="el-dialog" style="margin-top: 15vh;">
          <div class="el-dialog__header"><span class="el-dialog__title">group name Access Request</span><button type="button" aria-label="Close" class="el-dialog__headerbtn"><i class="el-dialog__close el-icon el-icon-close"></i></button></div>
          <!---->
          <!---->
        </div>
      </div>
    </transition-stub>
    <div class="header-row">
      <div class="header-names">
        <h1>group name</h1>
        <h2 class="ml-3 text-gray-600">
          (groupShortName)
        </h2>
      </div>
      <div class="header-buttons">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
    <div class="with-badges el-tabs el-tabs--top">
      <div class="el-tabs__header is-top">
        <div class="el-tabs__nav-wrap is-top">
          <div class="el-tabs__nav-scroll">
            <div role="tablist" class="el-tabs__nav is-top" style="transform: translateX(-0px);">
              <div class="el-tabs__active-bar is-top" style="width: 0px; transform: translateX(NaNpx); webkit-transform: translateX(NaNpx);"></div>
              <div id="tab-description" aria-controls="pane-description" role="tab" tabindex="-1" class="el-tabs__item is-top">Description</div>
              <div id="tab-datasets" aria-controls="pane-datasets" role="tab" tabindex="-1" class="el-tabs__item is-top">Datasets (4)</div>
              <div id="tab-members" aria-controls="pane-members" role="tab" tabindex="-1" class="el-tabs__item is-top"><span>
          Members (2)
          <!----></span></div>
              <div id="tab-databases" aria-controls="pane-databases" role="tab" tabindex="-1" class="el-tabs__item is-top"><span>
            Databases (2)
          </span></div>
              <div id="tab-settings" aria-controls="pane-settings" role="tab" aria-selected="true" tabindex="0" class="el-tabs__item is-top is-active">Settings</div>
            </div>
          </div>
        </div>
      </div>
      <div class="el-tabs__content">
        <!---->
        <!---->
        <!---->
        <!---->
        <div role="tabpanel" id="pane-settings" aria-labelledby="tab-settings" class="el-tab-pane">
          <div class="group-settings">
            <div class="header-row">
              <h2>Group Details</h2>
              <div class="flex-spacer"></div>
              <div class="header-row-buttons"><button type="button" class="el-button el-button--primary">
                  <!---->
                  <!----><span>
        Save
      </span></button></div>
            </div>
            <form class="el-form el-form--label-top">
              <div>
                <div class="el-form-item name is-required"><label for="name" class="el-form-item__label">Full name</label>
                  <div class="el-form-item__content">
                    <div class="el-input">
                      <!----><input type="text" autocomplete="off" max-length="50" class="el-input__inner">
                      <!---->
                      <!---->
                      <!---->
                      <!---->
                    </div>
                    <transition-stub name="el-zoom-in-top">
                      <!---->
                    </transition-stub>
                  </div>
                </div>
                <div class="el-form-item shortName is-required"><label for="shortName" class="el-form-item__label"><span>
        Short name
        <mock-el-popover trigger="hover" placement="right"><div slot-key="default"> <p>The short name will be shown whenever space is limited.</p></div><div slot-key="reference"><i class="el-icon-question metadata-help-icon"></i></div></mock-el-popover></span></label>
                  <div class="el-form-item__content">
                    <div class="el-input">
                      <!----><input type="text" autocomplete="off" max-length="20" class="el-input__inner">
                      <!---->
                      <!---->
                      <!---->
                      <!---->
                    </div>
                    <transition-stub name="el-zoom-in-top">
                      <!---->
                    </transition-stub>
                  </div>
                </div>
              </div>
              <!---->
            </form>
            <div style="margin-bottom: 2em;">
              <h2>Custom URL</h2>
              <!---->
              <!---->
              <div>
                <p><a href="/group/00000000-1111-2222-3333-444444444444" class="router-link-active">
                    http://localhost/group/<span class="urlSlug">00000000-1111-2222-3333-444444444444</span></a></p>
                <p><a href="mailto:contact@metaspace2020.eu">Contact us</a> to set up a custom URL to showcase your group.</p>
              </div>
            </div>
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
